### ISSUE TITLE
Review indices of PDFs

### ISSUE THREAD

**Issue by jstac** [QuantEcon/meta#4]
I haven't looked at the indices of our PDFs in a long time.

Are they OK?  Is there any easy way to improve them?  All recommendations are appreciated.

**Comment by Harveyt47** [comment]
Hey @jstac you mentioned this to me a while ago. Since then I have been reading some of the lectures in the PDFs. I have not seen any issues regarding the indices. But I'll keep looking.

**Comment by jstac** [comment]
I don't see actually an index, now that I look:

https://python-intro.quantecon.org/_downloads/pdf/quantitative_economics_with_python.pdf

Am I missing something?

**Comment by Harveyt47** [comment]
Hey @jstac are you talking about the pdf version of this https://python.quantecon.org/index_toc.html

If so I can see it on page three

**Comment by jstac** [comment]
That's the table of contents.  I'm asking about the index.

**Comment by mmcky** [comment]
@jstac the `index` seems to be missing. @Harveyt47 can you research adding `index` back into LaTeX and we can make sure it is in https://github.com/QuantEcon/lecture-python.theme/blob/master/templates/latex.tpl

**Comment by mmcky** [comment]
actually it should be in [latex_book.tpl](https://github.com/QuantEcon/lecture-python.theme/blob/master/templates/latex_book.tpl).

@jstac `index` elements are needed int he individual pdf's right?

**Comment by AakashGfude** [comment]
In case of pdfs, we have actually set all the index files in exclude patterns. So, they are avoided when building pdfs.
Like, for example :-
https://github.com/QuantEcon/lecture-python-advanced/blob/master/conf.py#L462

**Comment by Harveyt47** [comment]
Thanks @AakashGfude. Would we also have to add `\usepackage{imakeidx}`, `\makeindex` and `\printindex` to [latex_bok.tpl](https://github.com/QuantEcon/lecture-python.theme/blob/master/templates/latex_book.tpl)?

**Comment by AakashGfude** [comment]
@Harveyt47 I don't think that will be necessary, as we create tex files for index like any other pages. We can give it a try by removing the entry of `index` from the config variable and seeing what the pdf output is. If it is behaving weird, then we can try using other packages.

**Comment by Harveyt47** [comment]
Awesome thanks @AakashGfude I'll give removing `index` a try and see what happens.

**Comment by mmcky** [comment]
This issue is now old. There are index items included in all book `pdf`. 

@jstac would you like me to organise a review of that `index` to see how it is lining up with the current content?

**Comment by mmcky** [comment]
@jstac the index elements are looking pretty good. 

I did notice that many lecture titles are being added. 

<img width="812" alt="Screenshot 2024-01-29 at 1 21 10 pm" src="https://github.com/QuantEcon/meta/assets/8263752/5faaf742-08da-4db4-ae55-f09c20252548">

coming from

```
# {index}`A Problem that Stumped Milton Friedman <single: A Problem that Stumped Milton Friedman>`
```

which I think should be removed. 

But that was the old issue I found. 

- [ ] remove `index` roles from lecture titles
